projects/bootloader/test/test_config.c

```diff
@@ -0,0 +1,171 @@
+}
+
+void test_config_get(void) {
+  // This test tests to see that if config_get sets the given
+  // pointer to config page 1
+  persist_init(&s_test_persist_storage_1, BOOTLOADER_CONFIG_PAGE_1_FLASH_PAGE, &s_test_config_1,
+               sizeof(s_test_config_1), false);
+  persist_ctrl_periodic(&s_test_persist_storage_1, false);
+
+  BootloaderConfig test_input_config = { 0 };
+  TEST_ASSERT_OK(config_get(&test_input_config));
+
+  TEST_ASSERT_EQUAL_MEMORY(&s_test_config_1, &test_input_config, sizeof(s_test_config_1));
+}
+
+void test_config_commit(void) {
+  // I need to create an input config to give config_commit();
+  // I use random values for the input_config
+  BootloaderConfig test_input_config = { .CRC32 = 1,
+                                         .controller_board_ID = 1,
+                                         .controller_board_name = { 'a', '\0' },
+                                         .project_present = true,
+                                         .project_name = { 'a', '\0' },
+                                         .project_info = { 'a', '\0' },
+                                         .git_version = { 'a', '\0' },
+                                         .app_code = 1,
+                                         .app_size = 1 };
+
+  // Test 1, testing if page 2 and page 1 are both the new page
+  config_commit(&test_input_config);
+
+  persist_init(&s_test_persist_storage_1, BOOTLOADER_CONFIG_PAGE_1_FLASH_PAGE, &s_test_config_1,
+               sizeof(s_test_config_1), false);
+  persist_ctrl_periodic(&s_test_persist_storage_1, false);
+
+  persist_init(&s_test_persist_storage_2, BOOTLOADER_CONFIG_PAGE_2_FLASH_PAGE, &s_test_config_2,
+               sizeof(s_test_config_2), false);
+  persist_ctrl_periodic(&s_test_persist_storage_2, false);
+
+  TEST_ASSERT_EQUAL_MEMORY(&s_test_config_1, &test_input_config, sizeof(s_test_config_1));
+  TEST_ASSERT_EQUAL_MEMORY(&s_test_config_2, &s_test_config_1, sizeof(s_test_config_2));
+
+  // Test 2, testing if page 1 is reset when the input config was corrupted
+  s_function_pointer_hook = &prv_change_config_page;
+  TEST_ASSERT_EQUAL(STATUS_CODE_INTERNAL_ERROR, config_commit(&test_input_config));
+  s_function_pointer_hook = NULL;
+
+  persist_init(&s_test_persist_storage_1, BOOTLOADER_CONFIG_PAGE_1_FLASH_PAGE, &s_test_config_1,
+               sizeof(s_test_config_1), false);
+  persist_ctrl_periodic(&s_test_persist_storage_1, false);
+
+  persist_init(&s_test_persist_storage_2, BOOTLOADER_CONFIG_PAGE_2_FLASH_PAGE, &s_test_config_2,
+               sizeof(s_test_config_2), false);
+  persist_ctrl_periodic(&s_test_persist_storage_2, false);
+
+  TEST_ASSERT_EQUAL_MEMORY(&s_test_config_2, &s_test_config_1, sizeof(s_test_config_2));
+}
+
+StatusCode TEST_MOCK(persist_init)(PersistStorage *persist, FlashPage page, void *blob,
+                                   size_t blob_size, bool overwrite) {
+  if (s_function_pointer_hook != NULL) (*s_function_pointer_hook)();
```

Comment:
You don't need to dereference the function pointer manually, `s_function_pointer_hook()` will do it automatically.